### PR TITLE
Feature/clean up warnings from db

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
BSP-812: https://tools.hmcts.net/jira/browse/BPS-812

### Change description ###
Added setting OCR validation warnings to null during envelope finalization

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java [hunk 1]
```diff
@@ -63,12 +63,12 @@ public void tearDown() {
...
 
     @Test
-    public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_data() {
+    public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_data_and_warnings() {
         // given
         Envelope envelope = envelope(
             "JURISDICTION1",
             Status.NOTIFICATION_SENT,
-            createScannableItemsWithOcrData(3)
+            createScannableItemsWithOcrDataAndWarnings(3)
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java]: I think you will need to have two test scenarios
One with ocr and warnings
One with only ocr

You won’t always have ocr with warnings.

Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java]: Have updated

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java [hunk 2]
```diff
@@ -81,14 +81,98 @@ public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_dat
...
+            .allMatch(item -> item.getOcrData() != null && item.getOcrValidationWarnings() == null);
+    }
+
+    @Test
+    public void finaliseEnvelope_should_not_update_other_envelopes_with_warnings() {
+        // given
+        Envelope envelope1 = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItems(3, null, createWarnings())
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java]: this its invalid scenario you can't have warnings without OCR

Comment (apereverzin) [reply src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java]: Thank you, fixed this.

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java [hunk 3]
```diff
@@ -81,14 +81,98 @@ public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_dat
...
         assertThat(finalisedEnvelope).isPresent();
         assertThat(finalisedEnvelope.get().getStatus()).isEqualTo(Status.COMPLETED);
-        assertThat(finalisedEnvelope.get().getScannableItems()).allMatch(item -> item.getOcrData() == null);
+        assertThat(finalisedEnvelope.get().getScannableItems())
+            .allMatch(item -> item.getOcrData() == null && item.getOcrValidationWarnings() == null);
     }
 
     @Test
-    public void finaliseEnvelope_should_not_update_other_envelopes() {
+    public void finaliseEnvelope_should_update_envelope_status_and_clear_all_warnings() {
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java]: `finaliseEnvelope_should_update_envelope_status_and_clear_ocr_data_and_warnings` ?

File: src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java [hunk 4]
```diff
@@ -81,14 +81,98 @@ public void finaliseEnvelope_should_update_envelope_status_and_clear_all_ocr_dat
...
         // given
-        Envelope envelope1 = envelope("JURISDICTION1", Status.NOTIFICATION_SENT, createScannableItemsWithOcrData(3));
-        Envelope envelope2 = envelope("JURISDICTION1", Status.NOTIFICATION_SENT, createScannableItemsWithOcrData(3));
+        Envelope envelope = envelope(
+            "JURISDICTION1",
+            Status.NOTIFICATION_SENT,
+            createScannableItems(3, null, createWarnings())
```
Comment (nitinprabhu) [inline src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/services/EnvelopeFinaliserServiceTest.java]: no ocr data but warnings ?

**Pull request merged**